jerry-core/api/jerry.c

```diff
@@ -100,11 +100,7 @@ static const char * const wrong_args_msg_p = "wrong type of argument";
 static inline void JERRY_ATTR_ALWAYS_INLINE
 jerry_assert_api_available (void)
 {
-  if (JERRY_UNLIKELY (!(JERRY_CONTEXT (status_flags) & ECMA_STATUS_API_AVAILABLE)))
-  {
-    /* Terminates the execution. */
-    JERRY_UNREACHABLE ();
-  }
+  JERRY_ASSERT (JERRY_CONTEXT (status_flags) & ECMA_STATUS_API_AVAILABLE);
```

Comment:
I think this is not equal to the old code. `JERRY_ASSERT` has no effect in release builds, but `JERRY_UNREACHABLE` has (with GNUC).

Reply:
Which is quite the point of the change. Why should there be an effect in release builds and in GCC only?